.github/workflows/pull_request.yaml

```diff
@@ -0,0 +1,52 @@
+name: pull_request
+
+on:
+  pull_request:
+
+jobs:
+  pytest:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python: ['3.6', '3.7', '3.8']
+    name: Testing ${{ github.repository }} with python ${{ matrix.python }}
+    steps:
+      - uses: actions/checkout@v2
+      - name: Setup python
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python }}
+
+      - name: Install dependencies and script
+        run: pip install -r requirements.txt -e .
+
+      - name: Install test dependencies
+        run: pip install -r test-requirements.txt
+
+      - name: Lint with flake8
+        run: flake8
+
+      # No tests yet, enable this later
+      # - name: Test with pytest
+      #   run: pytest
+  build:
+    needs: pytest
```

Comment:
why??

Reply:
Just to save compute power, if it doesn't pass linting it doesn't need to build the containers which is a more expensive operation